proto/proto-backplane-grpc/src/main/proto/deephaven/proto/application.proto

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (c) 2016-2021 Deephaven Data Labs and Patent Pending
+ */
+
+syntax = "proto3";
+
+package io.deephaven.proto.backplane.grpc;
+
+option java_multiple_files = true;
+option optimize_for = SPEED;
+
+import "deephaven/proto/ticket.proto";
+
+/*
+ * Allows clients to list fields that are accessible to them.
+ */
+service ApplicationService {
+
+  /*
+   * Request the list of the fields exposed via the worker.
+   *
+   * - The first received message contains all fields that are currently available
+   *   on the worker. None of these fields will be RemovedFields.
+   * - Subsequent messages modify the existing state. Fields are identified by
+   *   their ticket and may be replaced or removed.
+   */
+  rpc listFields(ListFieldsRequest) returns (stream FieldsChangeUpdate) {}
+
+}
+
+message ListFieldsRequest {
+  // Intentionally empty and is here for backwards compatibility should this API change.
+}
+
+/*
+ * Represents a batch of fields.
+ */
+message FieldsChangeUpdate {
+  repeated FieldInfo fields = 1;
+}
+
+/*
+ * A lightweight object describing the exposed field.
+ */
+message FieldInfo {
```

Comment:
@mofojed any chance you want an icon field here? easy to add later, but to differentiate different plots, different "custom" things.

Reply:
Also, why isn't this a table? wouldnt it make future filtering easier, sorting, eventual access control...